portfolio/src/main/webapp/blog.html

```diff
@@ -41,38 +42,59 @@ <h4>Software Engineering Intern</h4>
 
         <div class="menu">
           <ul>
-              <li class="tab"><a href="/index.html#home"><i class="fas fa-home"></i>Home</a></li>
-              <li class="tab"><a href="/index.html#education"><i class="fas fa-user"></i>Education</a></li>
-              <li class="tab"><a href="/index.html#experience"><i class="fas fa-address-card"></i>Experience</a></li>
-              <li class="tab"><a href="/index.html#projects"><i class="fas fa-project-diagram"></i>Projects</a></li>
-              <li class="tab"><a href="/index.html#blog" class="active"><i class="fas fa-blog"></i>Blog</a></li>
+            <li class="tab"><a href="/index.html#home"><i class="fas fa-home"></i>Home</a></li>
+            <li class="tab"><a href="/index.html#education"><i class="fas fa-user"></i>Education</a></li>
+            <li class="tab"><a href="/index.html#experience"><i class="fas fa-address-card"></i>Experience</a></li>
+            <li class="tab"><a href="/index.html#projects"><i class="fas fa-project-diagram"></i>Projects</a></li>
+            <li class="tab"><a href="/index.html#blog" class="active"><i class="fas fa-blog"></i>Blog</a></li>
           </ul>
         </div>
     </div>
 
     <!-- Main Content Section -->
-
     <div class="menu">
       <section class="banner">
         <div class="info">
-          <h2>Blog Title</h2>
-          <p> Insert some blog post content here. </p>
+          <h2 id="title-container">Blog Title</h2>
+          <div id="post-container"></div>
           <h2>All Comments</h2>
           <div id="comment-container">
             <ul id="comment-list"></ul>
           </div>
-          <h2>Leave A Reply</h2>
+          <h2>Leave a Reply</h2>
           <form action="/data" method="POST">
-            <div class="inputBox">
+            <div class="input-box">
               <label for="name">Name</label>
               <input id="name" type="text" name="name" required>
             </div>
-            <div class="inputBox">
+            <div class="input-box">
               <label for="feedback">Comment</label>
               <textarea id="feedback" name="text-input" cols="100" rows="8" maxlength="10000" required></textarea>
             </div>
+            <div class="input-box">
+              <input type="hidden" name="current-url"  id="current-url" value="">
```

Comment:
Putting aside the issues of depending on the URL to store the tag for you...

It looks like the URL is being sent the server primarily to extract the tag. How about you only send the tag to the server? For the redirect, you already know where it should go - /blog.html?tag=TAG_GOES_HERE.